langchain/src/vectorstores/vectara.ts

```diff
@@ -102,11 +102,19 @@ export class VectaraStore extends VectorStore {
     this.apiKey = apiKey;
 
     const corpusId =
-      args.corpusId ?? getEnvironmentVariable("VECTARA_CORPUS_ID");
+      args.corpusId ??
+      getEnvironmentVariable("VECTARA_CORPUS_ID")
+        ?.split(",")
+        .map((id) => Number(id));
     if (!corpusId) {
       throw new Error("Vectara corpus id is not provided.");
     }
-    this.corpusId = corpusId;
+
+    if (typeof corpusId === "number") {
+      this.corpusId = [corpusId];
+    } else {
+      this.corpusId = corpusId;
```

Comment:
This could still be an empty array if `process.env.VECTARA_CORPUS_ID === ""` though right? Or if the env var was set to something not a number?

Happy to merge anyway if you're not concerned about it

Reply:
Sorry I completely missed this, should be fix now